src/BoundsInference.cpp

```diff
@@ -369,9 +369,9 @@ class BoundsInference : public IRMutator {
             // Check if it succeeded
             string result_name = unique_name('t');
             Expr result = Variable::make(Int(32), result_name);
-            vector<Expr> error_message = vec<Expr>("Bounds inference call to external func " + extern_name +
-                                                   " returned non-zero value:", result);
-            Stmt check = AssertStmt::make(EQ::make(result, 0), error_message);
+            Expr error = Call::make(Int(32), "halide_error_bounds_inference_call_failed",
```

Comment:
I thought about that, but I don't like lifting out constants that don't
have multiple uses, and these don't seem likely to acquire new uses in the
compiler source.

Each of these names are used in precisely two places: Once as source in the
runtime, and once as a constant string in the compiler source. I can't
think of any way to share the name between those two places in a way where
mismatches are caught earlier than getting the name wrong with the current
method.

...

Actually one possibility would be a mapping in HalideRuntime.h from error
codes to those names as const strings. Would such a mapping be generally
useful?

On Fri, Apr 3, 2015 at 2:19 PM, Steven Johnson <notifications@github.com>
wrote:

> In src/BoundsInference.cpp
> <https://github.com/halide/Halide/pull/730#discussion_r27756631>:
>
> > @@ -369,9 +369,9 @@ class BoundsInference : public IRMutator {
> >              // Check if it succeeded
> >              string result_name = unique_name('t');
> >              Expr result = Variable::make(Int(32), result_name);
> > -            vector<Expr> error_message = vec<Expr>("Bounds inference call to external func " + extern_name +
> > -                                                   " returned non-zero value:", result);
> > -            Stmt check = AssertStmt::make(EQ::make(result, 0), error_message);
> > +            Expr error = Call::make(Int(32), "halide_error_bounds_inference_call_failed",
>
> Seems like the call names should be constants somewhere rather than string
> literals.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/730/files#r27756631>.
>